### ISSUE TITLE
Dolt log -n 10 returns wrong results. Dolt log produces correct result.

### ISSUE THREAD

**Issue by timsehn** [liquidata-inc/dolt#420]
This affects DoltHub as well as it seems to use the -n code path behind the scenes.

```
Timothys-MacBook-Pro:corona-virus timsehn$ dolt branch
  add-views                                     	
  archived                                      	
* master                                        	
  modify-current-view                           	
  saved-dolt-diff-queries  
Timothys-MacBook-Pro:corona-virus timsehn$ dolt log | head -50
commit 1utsl6q1tvamuo7bhsikcaveudhkp156
Author: Tim Sehn <tim@liquidata.co>
Date:   Mon Feb 24 13:00:08 -0800 2020

	Added sample queries from blog entry manually because I stomped on the merge

commit qm9q2acrgdrrllvgjjc2vmtridvf74fu
Merge: noiqouncu4m3gmir746lc88lpkmqqsjc u8s6l7trfr0mlv506ppn7to281knmr4l
Author: Tim Sehn <tim@liquidata.co>
Date:   Mon Feb 24 12:21:12 -0800 2020

	Added new asample queries

commit u8s6l7trfr0mlv506ppn7to281knmr4l
Author: Zach Musgrave <zach@liquidata.co>
Date:   Mon Feb 24 12:19:14 -0800 2020

	Added mortality saved query

commit o05ur04b8n5qi4r77dp47uasqio0g24g
Author: LiquidataSystemAccount <liquidatasys@gmail.com>
Date:   Mon Feb 24 10:15:39 -0800 2020

	Automated import of new data downloaded from https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/ at Mon Feb 24 18:15:39 2020 GMT

commit l3h6tama110pmbluiv3jacrnumjettns
Author: LiquidataSystemAccount <liquidatasys@gmail.com>
Date:   Sun Feb 23 09:08:17 -0800 2020

	Automated import of new data downloaded from https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/ at Sun Feb 23 17:08:17 2020 GMT

commit phg9a8og7ge1nq6ogb6691podhrva26l
Author: LiquidataSystemAccount <liquidatasys@gmail.com>
Date:   Sat Feb 22 08:41:03 -0800 2020

	Automated import of new data downloaded from https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/ at Sat Feb 22 16:41:03 2020 GMT

commit o0kps99crujnum0fc2joiebii3r135gk
Author: LiquidataSystemAccount <liquidatasys@gmail.com>
Date:   Fri Feb 21 02:03:04 -0800 2020

	Automated import of new data downloaded from https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/ at Fri Feb 21 10:03:04 2020 GMT

commit tacg1c03h9t1731vcansgq2qpelhia9u
Author: LiquidataSystemAccount <liquidatasys@gmail.com>
Date:   Thu Feb 20 05:47:51 -0800 2020

	Automated import of new data downloaded from https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/ at Thu Feb 20 13:47:51 2020 GMT

commit ch4g82ip0bdnuefdlfo6m7nk7jt4decs
Timothys-MacBook-Pro:corona-virus timsehn$ dolt log -n 10
commit 1utsl6q1tvamuo7bhsikcaveudhkp156
Author: Tim Sehn <tim@liquidata.co>
Date:   Mon Feb 24 13:00:08 -0800 2020

	Added sample queries from blog entry manually because I stomped on the merge

commit qm9q2acrgdrrllvgjjc2vmtridvf74fu
Merge: noiqouncu4m3gmir746lc88lpkmqqsjc u8s6l7trfr0mlv506ppn7to281knmr4l
Author: Tim Sehn <tim@liquidata.co>
Date:   Mon Feb 24 12:21:12 -0800 2020

	Added new asample queries

commit noiqouncu4m3gmir746lc88lpkmqqsjc
Author: Tim Sehn <tim@liquidata.co>
Date:   Tue Feb 18 17:24:02 -0800 2020

	Added a saved query for last to automated commits

commit vrnkl5ouvsk6412gibtsghbvr8o400es
Author: Tim Sehn <tim@liquidata.co>
Date:   Tue Feb 18 17:20:11 -0800 2020

	Added a saved query for dolt_diff_cases table

commit qf1r6htjdnkqlucsmin72p9c9fqo0t7h
Author: Tim Sehn <tim@liquidata.co>
Date:   Tue Feb 18 11:31:44 -0800 2020

	Changed display order of sample queries

commit a7rhnn3kp6pobi0ompb3du8tcneh3p55
Author: Tim Sehn <tim@liquidata.co>
Date:   Tue Feb 18 11:22:39 -0800 2020

	Added sample queries

commit f4ucpbi9eeo4ebklql8b13lo5r6924pu
Author: Tim Sehn <tim@liquidata.co>
Date:   Tue Feb 18 09:21:43 -0800 2020

	Automated import of new data downloaded from https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/ at Tue Feb 18 17:21:43 2020 GMT

commit 2nv1qb24s58dbhtv69dfbn8dri06a218
Author: Tim Sehn <tim@liquidata.co>
Date:   Mon Feb 17 15:06:17 -0800 2020

	Add time_series view

commit 23219cpmod4qcakdv6bkg4va4tqmq9td
Author: Tim Sehn <tim@liquidata.co>
Date:   Mon Feb 17 14:49:19 -0800 2020

	Changed title of README

commit tu3b4f3cgrcu9mjkk5f87mfhhnkucslv
Author: Tim Sehn <tim@liquidata.co>
Date:   Mon Feb 17 12:48:56 -0800 2020

	Updated current view to only show the date (not the time) as new data is all imported at 00:00:00 UTC
```

**Comment by timsehn** [comment]
This is the difference between topological sort of commits and date-based. We should make them the same at least.

**Comment by timsehn** [comment]
Skipped bats test in #424

**Comment by timsehn** [comment]
@bheni fixed this in #426
